libs/core/datastructures/include/hpx/datastructures/detail/small_vector.hpp

```diff
@@ -48,9 +48,11 @@ namespace hpx::detail {
             return this == &rhs;
         }
 
+        // Note: the PMR infrastructure calls this with the number of bytes to
+        // allocate but the underlying allocator expects the number of elements
         void* do_allocate(std::size_t count, std::size_t) override
         {
-            return allocator_.allocate(count);
+            return allocator_.allocate((count + sizeof(T) - 1) / sizeof(T));
```

Comment:
This is the key change, right?

Reply:
This is not the actual problem but reduces the memory allocation (if it happens) to a minimum. Before it was allocating too much memory on expansion.

Reply:
Ah, right. This actually reduces the amount of memory allocated.